modules/navier_stokes/include/fvbcs/NSFVHeatFluxBC.h

```diff
@@ -0,0 +1,99 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FVFluxBC.h"
+#include "NSEnums.h"
+
+/**
+ * This boundary condition sets a constant heat flux with a splitting between the
+ * fluid and solid phases according to one of
+ *
+ * - local value of porosity
+ * - local value of thermal conductivity
+ * - local value of effective thermal conductivity
+ * - domain-averaged value of porosity
+ * - domain-averaged value of thermal conductivity
+ * - domain-averaged value of effective thermal conductivity
+ *
+ * Local values are obtained on the boundary, while domain-averaged, or _global_, values
+ * are computed as averages over the domain. This boundary condition does not specify the
+ * interpretation of what constitutes the domain, so you may choose for the global values to
+ * represent averages over the entire geometry, or possibly a finite-width region near the wall.
+ *
+ * For instance, if the heat flux is split according to the local value of thermal
+ * conductivity, the heat flux entering the fluid phase is
+ * \f$\frac{k_f}{k_f+k_s}\dot{q}\f$, where \f$k_f\f$ is the fluid thermal conductivity,
+ * \f$k_s\f$ is the solid thermal conductivity, and \f$\dot{q}\f$ is the total value of
+ * the heat flux.
+ *
+ * Neither selection of a local or global splitting is perfect.
+ */
+class NSFVHeatFluxBC : public FVFluxBC
+{
+public:
+  static InputParameters validParams();
+  NSFVHeatFluxBC(const InputParameters & parameters);
+
+protected:
+  virtual ADReal computeQpResidual() override;
+
+  /// Value of the heat flux
+  const Real & _value;
+
+  /// Which phase this boundary condition is applied to, i.e. 'fluid' or 'solid'
+  const NS::phase::PhaseEnum _phase;
```

Comment:
Sigh. [General style guide](https://google.github.io/styleguide/cppguide.html#Namespace_Names) is to use lower case for namespaces which these new namespaces are doing. I wish that I had read the style guide earlier...but basically all of our namespaces to date have been capitalized. I don't love the mix of different capitalization but I'm also not going to say we shouldn't follow generally accepted styles for new namespaces. So in conclusion... I'm fine with this